### DISCUSSION TITLE
Extracting decompressed audio samples?

### DISCUSSION THREAD

**Discussion by imaginary-person** [libsndfile/libsndfile#722] [Q&A] [answered]
Hello,

I want to learn how digital sound works in general.
I've only read Mozilla's guide on [digital audio](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Audio_concepts) so far.

Can this library be used to extract decompressed audio samples from audio files that were encoded by supported codecs?

Please give me some advice on getting started. Thank you!

**Comment by evpobr** [comment, marked as answer]
Hi @imaginary-person .

Yes, libsndfile can extract audio samples from any supported format. Start from [our documentation](http://libsndfile.github.io/libsndfile/), especially [Programming interface](http://libsndfile.github.io/libsndfile/api.html).

**Comment by imaginary-person** [comment]
Hello @evpobr, thanks for your help!
